libbeat/kibana/index_pattern_generator.go

```diff
@@ -29,33 +29,35 @@ func NewGenerator(indexName, beatName, beatDir, version string) (*IndexPatternGe
 	}
 
 	return &IndexPatternGenerator{
-		indexName:        indexName,
-		version:          version,
-		fieldsYaml:       fieldsYaml,
-		targetDirDefault: createTargetDir(beatDir, "default"),
-		targetDir5x:      createTargetDir(beatDir, "5.x"),
-		targetFilename:   beatName + ".json",
+		indexName:      indexName,
+		fieldsYaml:     fieldsYaml,
+		beatVersion:    beatVersion,
+		version:        version,
+		targetDir:      createTargetDir(beatDir, version),
+		targetFilename: beatName + ".json",
 	}, nil
 }
 
 // Create the Index-Pattern for Kibana for 5.x and default.
-func (i *IndexPatternGenerator) Generate() ([]string, error) {
+func (i *IndexPatternGenerator) Generate() (string, error) {
 	commonFields, err := common.LoadFieldsYaml(i.fieldsYaml)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	index5xPath, err := i.generate5x(commonFields)
-	if err != nil {
-		return nil, err
+	var path string
+
+	if i.version.Major == 5 {
+		path, err = i.generate5x(commonFields)
+	} else {
+		path, err = i.generate6x(commonFields)
 	}
 
-	index6xPath, err := i.generate6x(commonFields)
 	if err != nil {
-		return nil, err
+		return "", err
```

Comment:
Since you already define the path veriable upfront you could `return path, err` without an explicit check here.

Reply:
good point, changed